hexrd/instrument.py

```diff
@@ -3576,13 +3580,20 @@ def unwrap_dict_to_h5(grp, d, asattr=False):
             unwrap_dict_to_h5(subgrp, item, asattr=asattr)
         else:
             if asattr:
-                grp.attrs.create(key, item)
+                try:
+                    grp.attrs.create(key, item)
+                except(TypeError):
+                    if item is None:
+                        continue
```

Comment:
Could we either check if the item is `None` like the following:
```suggestion
                if item is None:
                    continue
                grp.attrs.create(key, item)
```

Or add an `else: raise` to the end of this (see below)? 

```suggestion
                try:
                    grp.attrs.create(key, item)
                except(TypeError):
                    if item is None:
                        continue
                    else:
                        raise
```

If the `item` is not `None`, then we probably want to propagate the exception so that we see it. If we don't, writing the item will silently fail, and might produce an invalid HDF5 file without us knowing.

Adding a `raise` in exception handling will propagate the exception.

Reply:
yes -- I agree with the second and applied it.